service/installer/spec/spec.go

```diff
@@ -13,7 +13,11 @@ type Installer interface {
 	// Install takes a Project, and installs the referenced chart. If an error
 	// occurs, the returned error will be non-nil.
 	Install(project Project) error
-	// List takes a list of projects and returns a list of installed charts in
-	// form of a list of Project items that match the given list.
-	List(projects []Project) ([]Project, error)
+	// List returns a list of installed charts in form of a list of Project items
+	// that match the given list.
+	//
+	// NOTE that the shas/refs of projects returned here are eventually
+	// incomplete. This is because of certain helm limitations when listing
+	// charts.
+	List() ([]Project, error)
```

Comment:
The current problem with the update is that when the sha gets updated we cannot guarantee to find the current state based on the information that are not available anymore, because the sha in the current custom object changed. In the first place we deployed sha AAA. After some time we receive an update and are asked to deploy sha BBB. Now we do not know about sha AAA for sure. We only know about sha AA because helm does not fully list the complete sha. Thus we have to expose this insufficiency for now and let the project resource deal with it.

Reply:
The only way I see to really solve this is to use the tiller grpc api

Reply:
+1 #later